### PR TITLE
Batches and batches of youtube downloading

### PR DESCRIPTION
**Pull request by abMatGit**
This should handle youtube playlists and download each song asynchronously !

Kudos to the awesome async work already done to dispatch the youtube-dl instructions and handling them in parallel 💯 💯 

Front-end side of the PR: https://github.com/alan-andrade/detektor/pull/2

### CODE REVIEW

File: lib/task.ex [hunk 1]
```diff
@@ -9,13 +9,34 @@ defmodule Detektor.Task do
...
+  def fetchPlaylist(url, parent, _) do
+    cmd = "youtube-dl"
+    args = [ url, "--flat-playlist", "-J"]
+
+    Logger.debug"> youtube-dl #{Enum.join(args, " ")}"
+    case System.cmd(cmd, args, [parallelism: true, stderr_to_stdout: true]) do
+      {output, 0} ->
+        {:ok, youtubeHash}  = output |> String.split("\n") |> hd |> Poison.decode
+
+        constructYoutubeURL = fn(id) -> "https://www.youtube.com/watch?v=#{id}" end
```
Comment (alan-andrade) [inline lib/task.ex]: This is constraining us to only youtube. I wonder if we can get the full url instead of only the track ID.

**Comment by alan-andrade** [comment]
This is 🔥 🔥 🔥  !

The only important comment is that we should try to use youtube-dl greatness and not bound ourselves into only youtube.

We should merge and tackle it in another PR.

This is pim 👌  🙏 

https://www.youtube.com/watch?v=oq2Oq1J6t70

**Comment by alan-andrade** [comment]
> sidenote: we (both) need more elixir foo, the code is a litttttttle bit hard to follow. It might just be that we need to get acquainted. (again)

**Comment by abMatGit** [comment]
Thanks for mentioning about restricting the batches to just youtube, its totally unnecessary! 🍻 🍻 

It turns out doing a playlist JSON command-line **DOES** return URLs but i was confused because for youtube it returned it in the form: 

```
aMat@local ➜ ~  youtube-dl --flat-playlist -j <youtube_playlist>
{"url": "iNPQx_Bb2Fo", "_type": "url", "ie_key": "Youtube", "id": "iNPQx_Bb2Fo", "title": "Fleetwood Mac - You Make Loving Fun"}
{"url": "6ul-cZyuYq4", "_type": "url", "ie_key": "Youtube", "id": "6ul-cZyuYq4", "title": "Fleetwood Mac - Go Your Own Way (HQ)"}
```

The urls look like ids BUUUUUUTTTTTT youtube-dl is smart enough to know that these URLs are from **youtube** so you can download it just fine with `youtube-dl <insert id>`. 💥 💥 ❗️ ❗️ 

So for soundcloud playlists it looks like:
```
aMat@local ➜ ~  youtube-dl --flat-playlist -j <soundcloud_playlist>               
{"url": "http://soundcloud.com/agents-of-time/polina-master-a", "_type": "url", "ie_key": "Soundcloud", "id": "148556338"}
{"url": "http://soundcloud.com/correspondant/magdalene", "_type": "url", "ie_key": "Soundcloud", "id": "177509332"}
{"url": "http://soundcloud.com/musumeci/gemini_vi", "_type": "url", "ie_key": "Soundcloud", "id": "148837846"}
```

Long story short, we can handle all platforms AND kill some code! 🔪 🔪 👍 👍 🔥 🔥 🚒

File: lib/task.ex (hunk 1 above)
Comment (abMatGit) [reply lib/task.ex]: 🔪

**Comment by abMatGit** [comment]
I think to have good elixir foo, or to make elixir code "look good" you really need to utilize functional programming and utilize its piping syntax. 

I think what reads very nicely is following the flow of steps in a pipe progression if everything is well named.
